### ISSUE TITLE
Sql federation does not support the `is` function

### ISSUE THREAD

**Issue by FlyingZC** [apache/shardingsphere#26567]
### Which version of ShardingSphere did you use?
Master.

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
Proxy.

### Expected behavior
Execute sql successfully.

### Actual behavior
```sql
SQLSTATE[HY000]: General error: 30000 Unknown exception: Unsupported SQL operator: `IS` The SQL being executed was: /* SHARDINGSPHERE_HINT: SHARDING_DATABASE_VALUE= 12538 */ SELECT `kcu`.`CONSTRAINT_NAME` AS `constraint_name`, `kcu`.`COLUMN_NAME` AS `column_name`, `kcu`.`REFERENCED_TABLE_NAME` AS `referenced_table_name`, `kcu`.`REFERENCED_COLUMN_NAME` AS `referenced_column_name` FROM `information_schema`.`REFERENTIAL_CONSTRAINTS` AS `rc` JOIN `information_schema`.`KEY_COLUMN_USAGE` AS `kcu` ON ( `kcu`.`CONSTRAINT_CATALOG` = `rc`.`CONSTRAINT_CATALOG` OR (`kcu`.`CONSTRAINT_CATALOG` IS NULL AND `rc`.`CONSTRAINT_CATALOG` IS NULL) ) AND `kcu`.`CONSTRAINT_SCHEMA` = `rc`.`CONSTRAINT_SCHEMA` AND `kcu`.`CONSTRAINT_NAME` = `rc`.`CONSTRAINT_NAME` WHERE `rc`.`CONSTRAINT_SCHEMA` = database() AND `kcu`.`TABLE_SCHEMA` = database() AND `rc`.`TABLE_NAME` = 'comment_configs' AND `kcu`.`TABLE_NAME` = 'comment_configs'
```

### Reason analyze (If you can)
Sql federation does not support the `is` function now.

**Comment by kanha-gupta** [comment]
@FlyingZC This PR :- https://github.com/apache/shardingsphere/pull/25564 solves the issue of "IS" operator  for IS NULL and IS NOT NULL operators. Please inform if further work is needed :)

**Comment by strongduanmu** [comment]
Fixed by #25564, thank you for your feedback. @kanha-gupta
